src/validation/ValidationExecutor.ts

```diff
@@ -270,6 +270,25 @@ export class ValidationExecutor {
                     this.execute(subValue, targetSchema, validationError.children);
                 });
 
+            } else if (value instanceof Set) {
+                let index = 0;
+                value.forEach((subValue: any) => {
+                    const validationError = this.generateValidationError(value, subValue, index.toString());
+                    errors.push(validationError);
+
+                    this.execute(subValue, targetSchema, validationError.children);
+
+                    ++index;
+                });
+
+            } else if (value instanceof Map) {
```

Comment:
Looks like duplicate code. Should be in first statement:
```ts
if (value instanceof Array || value instanceof Map) {
```

Reply:
As [you can see](https://travis-ci.org/typestack/class-validator/jobs/559511528), it is not the best idea to merge these calls...